Modules/mathmodule.c

```diff
@@ -2490,6 +2490,151 @@ math_isclose_impl(PyObject *module, double a, double b, double rel_tol,
 }
 
 
+/*[clinic input]
+math.combinations
+
+    n: object
+
+    k: object
+    /
+
+Return the binomial coefficient indexed by the pair of integers n >= k >= 0.
+
+It is the coefficient of kth term in polynomial expansion of the expression
+(1 + x)^n. It is also known as the number of ways to choose an unordered
+subset of k elements from a fixed set of n elements, usually called
+*n choose k*.
+
+Raises a TypeError if argument(s) are non-integer and ValueError
+if argument(s) are negative or k > n.
+
+[clinic start generated code]*/
+
+static PyObject *
+math_combinations_impl(PyObject *module, PyObject *n, PyObject *k)
+/*[clinic end generated code: output=19e8a448a1be51e0 input=a1334a4fbeb72a00]*/
+{
+    if (!(PyLong_Check(n) && PyLong_Check(k))) {
+        PyErr_SetString(PyExc_TypeError,
+            "combinations() only accepts integer arguments");
+        return NULL;
+    }
+    n = PyNumber_Long(n);
+    if (n == NULL) {
+        return NULL;
+    }
+    k = PyNumber_Long(k);
+    if (k == NULL) {
+        Py_DECREF(n);
+        return NULL;
+    }
+
+    PyObject *val = NULL,
+        *temp_obj1 = NULL,
+        *temp_obj2 = NULL,
+        *dump_var = NULL;
+    int overflow, cmp;
+    long long i, terms;
+
+    cmp = PyObject_RichCompareBool(n, k, Py_LT);
+    if (cmp == 1) {
+        PyErr_Format(PyExc_ValueError,
+                     "n must be an integer greater or equal to k");
+        goto fail_comb;
+    }
+    else if (cmp == -1) {
+        goto fail_comb;
+    }
+
+    /* b = min(b, a - b) */
+    dump_var = PyNumber_Subtract(n, k);
+    if (dump_var == NULL) {
+        goto fail_comb;
+    }
+    cmp = PyObject_RichCompareBool(k, dump_var, Py_GT);
+    if (cmp == 1) {
+        Py_DECREF(k);
+        k = dump_var;
+        dump_var = NULL;
+    }
+    else if (cmp == -1) {
+        goto fail_comb;
+    }
+    else {
+        Py_DECREF(dump_var);
+        dump_var = NULL;
+    }
+
+    terms = PyLong_AsLongLongAndOverflow(k, &overflow);
+    if (terms == -1 && PyErr_Occurred()) {
+        goto fail_comb;
+    }
+    else if (overflow == 1) {
+        PyErr_Format(PyExc_OverflowError,
+                     "(n - k) and k must not exceed %lld",
```

Comment:
It has to be minimum of the two since the algorithm uses symmetry to reduce no of iterations

Reply:
Ah, right.